tests/unit/DetectorTest.php

```diff
@@ -0,0 +1,461 @@
+                'framework' => 'Astro',
+                'installCommand' => 'npm install',
+                'buildCommand' => 'npm run build',
+                'outputDirectory' => './dist',
+            ],
+            [
+                'files' => ['app', 'public', 'remix.config.js', 'remix.env.d.ts', 'sandbox.config.js', 'tsconfig.json', 'package.json'],
+                'framework' => 'Remix',
+                'installCommand' => 'npm install',
+                'buildCommand' => 'npm run build',
+                'outputDirectory' => './build',
+            ],
+            [
+                'files' => ['src', 'static', 'scripts', 'eslint.config.js', 'package.json', 'pnpm-lock.yaml', 'svelte.config.js', 'tsconfig.js', 'vite.config.js', 'vite.config.lib.js'],
+                'framework' => 'SvelteKit',
+                'installCommand' => 'npm install',
+                'buildCommand' => 'npm run build',
+                'outputDirectory' => './build',
+            ],
+            [
+                'files' => ['src', 'types', 'makefile', 'components.js', 'debug.js', 'package.json', 'svelte.config.js'],
+                'framework' => 'SvelteKit',
+                'installCommand' => 'npm install',
+                'buildCommand' => 'npm run build',
+                'outputDirectory' => './build',
+            ],
+            [
+                'files' => ['app', 'backend', 'public', 'Dockerfile', 'docker-compose.yml', 'ecosystem.config.js', 'middleware.ts', 'next.config.js', 'package-lock.json', 'package.json', 'server.js', 'tsconfig.json'],
+                'framework' => 'Next.js',
+                'installCommand' => 'npm install',
+                'buildCommand' => 'npm run build',
+                'outputDirectory' => './.next',
+            ],
+            [
+                'files' => ['assets', 'components', 'layouts', 'pages', 'babel.config.js', 'error.vue', 'nuxt.config.js', 'yarn.lock'],
+                'framework' => 'Nuxt',
+                'installCommand' => 'npm install',
+                'buildCommand' => 'npm run build',
+                'outputDirectory' => './output',
+            ],
+            // test for FAILURE
+            [
+                'files' => ['index.html', 'style.css'],
+                'framework' => null,
+            ],
+        ];
+    }
+
+    public function test_rendering_detection(): void
+    {
+        foreach ($this->renderingDataProvider() as $data) {
+            $detector = new Rendering($data['files'], $data['framework']);
+            $detector
+                ->addOption(new SSR)
+                ->addOption(new SSG);
+
+            $rendering = $detector->detect();
+
+            if ($rendering) {
+                $this->assertEquals($data['rendering'], $rendering->getName());
+            } else {
+                $this->assertEquals($data['rendering'], null);
+            }
+        }
+    }
+
+    public function renderingDataProvider(): array
```

Comment:
All methods in here should start with word `test`, otherwise they wont get ran by PHPUnit. I think.